prototype_source/lite_interpreter.rst

```diff
@@ -0,0 +1,223 @@
+    #include "caffe2/utils/string_utils.h"
+    #include "torch/csrc/autograd/grad_mode.h"
+    #include "torch/script.h"
+    #include <torch/csrc/jit/mobile/function.h>
+    #include <torch/csrc/jit/mobile/import.h>
+    #include <torch/csrc/jit/mobile/interpreter.h>
+    #include <torch/csrc/jit/mobile/module.h>
+    #include <torch/csrc/jit/mobile/observer.h>
+
+.. code-block:: swift
+
+    @implementation TorchModule {
+    @protected
+    // torch::jit::script::Module _impl;
+     torch::jit::mobile::Module _impl;
+    }
+
+    - (nullable instancetype)initWithFileAtPath:(NSString*)filePath {
+      self = [super init];
+      if (self) {
+          try {
+              _impl = torch::jit::_load_for_mobile(filePath.UTF8String);
+             //  _impl = torch::jit::load(filePath.UTF8String);
+             //  _impl.eval();
+            } catch (const std::exception& exception) {
+                NSLog(@"%s", exception.what());
+                return nil;
+            }
+        }
+        return self;
+    }
+
+
+5.2 Update ``ViewController.swift``
+
+.. code-block:: swift
+
+    //  if let filePath = Bundle.main.path(forResource:
+    //      "deeplabv3_scripted", ofType: "pt"),
+    //      let module = TorchModule(fileAtPath: filePath) {
+    //      return module
+    //  } else {
+    //      fatalError("Can't find the model file!")
+    //  }
+    if let filePath = Bundle.main.path(forResource:
+        "deeplabv3_scripted", ofType: "ptl"),
+        let module = TorchModule(fileAtPath: filePath) {
+        return module
+    } else {
+        fatalError("Can't find the model file!")
+    }
+
+6. Build and test the app in Xcode.
+
+How to use lite interpreter + custom build
+------------------------------------------
+1. Verify your PyTorch version is 1.4.0 or above. You can do that by checking the value of `torch.__version__`.
+
+2. To dump the operators in your model, say `deeplabv3_scripted`, run the following lines of Python code:
+
+.. code-block:: python
+
+    # Dump list of operators used by deeplabv3_scripted:
+    import torch, yaml
+    model = torch.jit.load('deeplabv3_scripted.ptl')
+    ops = torch.jit.export_opnames(model)
+    with open('deeplabv3_scripted.yaml', 'w') as output:
+        yaml.dump(ops, output)
+
+In the snippet above, you first need to load the ScriptModule. Then, use export_opnames to return a list of operator names of the ScriptModule and its submodules. Lastly, save the result in a yaml file.
+
+3. To run the build script locally with the prepared yaml list of operators, pass in the yaml file generate from the last step into the environment variable SELECTED_OP_LIST. Also in the arguments, specify BUILD_PYTORCH_MOBILE=1 as well as the platform/architechture type.
+
+**iOS**: Take the simulator build for example, the command should be:
+
+.. code-block:: bash
+
+   SELECTED_OP_LIST=deeplabv3_scripted.yaml BUILD_PYTORCH_MOBILE=1 IOS_PLATFORM=SIMULATOR BUILD_LITE_INTERPRETER=1 ./scripts/build_ios.sh
```

Comment:
Dont need `BUILD_PYTORCH_MOBILE=1` I believe

Reply:
Hmm for simulator as well? I've been using this cmd and it's also mentioned on the mobile page `https://pytorch.org/mobile/ios/#build-pytorch-ios-libraries-from-source`? If it's not needed, we probably also need to update the mobile page?